pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -883,35 +886,22 @@ func PruneEmpty(config map[string]interface{}) {
 	}
 }
 
-func addAddresses(secrets corecontrollers.SecretCache, config map[string]interface{}, machine *capi.Machine) {
-	if data := machine.Annotations[InternalAddressAnnotation]; data != "" {
-		config["node-ip"] = data
-	}
-
-	if data := machine.Annotations[AddressAnnotation]; data != "" {
-		config["node-external-ip"] = data
-		// Don't check for an IP address from the secret if it is already provided.
-		return
-	}
+func addAddresses(secrets corecontrollers.SecretCache, controlPlaneIPs []interface{}, config map[string]interface{}, machine *capi.Machine) {
+	internalIPAddress, ipAddress := getIPAddresses(secrets, machine)
+	setNodeExternalIP := ipAddress != "" && internalIPAddress != "" && ipAddress != internalIPAddress
 
-	if config["cloud-provider-name"] != nil || machine.Annotations["objectset.rio.cattle.io/id"] == "unmanaged-machine" {
-		// Unmanaged machines won't have a state secret,
-		// and the cloud provider will handle this, if it is set.
-		return
+	if setNodeExternalIP && !isOnlyWorker(machine) {
+		config["advertise-address"] = internalIPAddress
+		appendToSlice(config, "tls-san", controlPlaneIPs...)
```

Comment:
I'm not 100% convinced that tls-san needs to be all controlPlaneIPs.  What fails when you only set the current node IPs?

Reply:
I retested again and everything seems to work with only providing the external IP for the particular node.